rollup/rollup_sync_service/rollup_sync_service.go

```diff
@@ -509,69 +503,56 @@ func validateBatch(batchIndex uint64, event *L1FinalizeBatchEvent, parentFinaliz
-		daBatch, err := codecv2.NewDABatch(batch)
-		if err != nil {
-			return 0, nil, fmt.Errorf("failed to create codecv2 DA batch, batch index: %v, err: %w", batchIndex, err)
-		}
-		localBatchHash = daBatch.Hash()
-	} else if determinedCodecVersion == encoding.CodecV3 {
-		daBatch, err := codecv3.NewDABatch(batch)
+
+	codec, err := encoding.GetCodec(codecVersion)
+	if err != nil {
+		return 0, nil, fmt.Errorf("unsupported codec version: %v, batch index: %v, err: %w", codecVersion, batchIndex, err)
+	}
+
+	if codecVersion != encoding.CodecV4 {
+		daBatch, err := codec.NewDABatch(batch)
 		if err != nil {
-			return 0, nil, fmt.Errorf("failed to create codecv3 DA batch, batch index: %v, err: %w", batchIndex, err)
+			return 0, nil, fmt.Errorf("failed to create DA batch, batch index: %v, codec version: %v, err: %w", batchIndex, codecVersion, err)
 		}
 		localBatchHash = daBatch.Hash()
-	} else if determinedCodecVersion == encoding.CodecV4 {
-		// Check if committedBatchMeta exists, for backward compatibility with older client versions
+	} else {
 		if committedBatchMeta == nil {
-			return 0, nil, fmt.Errorf("missing committed batch metadata for codecV4, please use the latest client version, batch index: %v", batchIndex)
+			return 0, nil, fmt.Errorf("missing committed batch metadata, please use the latest client version, batch index: %v, codec version: %v", batchIndex, codecVersion)
 		}
 
-		// Validate BlobVersionedHashes
 		if committedBatchMeta.BlobVersionedHashes == nil || len(committedBatchMeta.BlobVersionedHashes) == 1 {
-			return 0, nil, fmt.Errorf("invalid blob hashes, batch index: %v, blob hashes: %v", batchIndex, committedBatchMeta.BlobVersionedHashes)
+			return 0, nil, fmt.Errorf("invalid blob hashes, batch index: %v, codec version: %v, blob hashes: %v", batchIndex, codecVersion, committedBatchMeta.BlobVersionedHashes)
 		}
 
-		// Attempt to create DA batch with compression
-		daBatch, err := codecv4.NewDABatch(batch, true)
-		if err != nil {
-			// If compression fails, try without compression
-			log.Warn("failed to create codecv4 DA batch with compress enabling", "batch index", batchIndex, "err", err)
-			daBatch, err = codecv4.NewDABatch(batch, false)
-			if err != nil {
-				return 0, nil, fmt.Errorf("failed to create codecv4 DA batch, batch index: %v, err: %w", batchIndex, err)
-			}
-		} else if daBatch.BlobVersionedHash != committedBatchMeta.BlobVersionedHashes[0] {
-			// Inconsistent blob versioned hash, fallback to uncompressed DA batch
-			log.Warn("impossible case: inconsistent blob versioned hash", "batch index", batchIndex, "expected", committedBatchMeta.BlobVersionedHashes[0], "actual", daBatch.BlobVersionedHash)
-			daBatch, err = codecv4.NewDABatch(batch, false)
+		// Try creating the DA batch with compression
+		codec.SetCompression(true)
+		daBatch, err := codec.NewDABatch(batch)
+
+		blobVersionHashes := committedBatchMeta.BlobVersionedHashes
+		if blobVersionHashes == nil || len(blobVersionHashes) == 1 {
```

Comment:
this already checked above in line 531.

Reply:
nice catch. it's a bug, this should be the value fetched from the DA batch. fixed in [bcb524f](https://github.com/scroll-tech/go-ethereum/pull/997/commits/bcb524f8c939cc8a01c6c0afa115de1a7d2a560c).